lib/toy/module_instance.rb

```diff
@@ -45,9 +45,11 @@ def instance_method(selector)
       included_modules.each do |nested_mixin|
         method = nested_mixin.instance_method(selector)
         return method if method
+      rescue ::NameError
+        nil
       end
 
-      if superclass
+      if self.class == Class && superclass
```

Comment:
In the hindsight provided by #38, it’s possible we could’ve achieved this more neatly by overriding `#instance_method` in `ClassInstance` rather than putting a conditional here. Oh well, it’s in the past. 😄

Reply:
🤦‍♀️ Hindsight is 20/20!